lms/djangoapps/discussion/tests/test_views.py

```diff
@@ -356,61 +356,72 @@ def setUp(self):
         # course is outside the context manager that is verifying the number of queries,
         # and with split mongo, that method ends up querying disabled_xblocks (which is then
         # cached and hence not queried as part of call_single_thread).
-        (ModuleStoreEnum.Type.mongo, 1, 5, 3, 13, 1),
-        (ModuleStoreEnum.Type.mongo, 50, 5, 3, 13, 1),
+        (ModuleStoreEnum.Type.mongo, False, 1, 5, 3, 13, 1),
+        (ModuleStoreEnum.Type.mongo, False, 50, 5, 3, 13, 1),
         # split mongo: 3 queries, regardless of thread response size.
-        (ModuleStoreEnum.Type.split, 1, 3, 3, 12, 1),
-        (ModuleStoreEnum.Type.split, 50, 3, 3, 12, 1),
+        (ModuleStoreEnum.Type.split, False, 1, 3, 3, 12, 1),
+        (ModuleStoreEnum.Type.split, False, 50, 3, 3, 12, 1),
+
+        # Enabling Enterprise integration increases the number of (cached and uncached) SQL queries by 1,
+        # because the presence of the user's consent for the course must be checked.
+        # But there should be no effect on the number of mongo queries made.
```

Comment:
Optional/opinion-heavy: IMHO this seems like an intrusive addition to this test code, for a pretty small risk that the enterprise check will decrease performance. I would prefer to just test unit the new decorator you added.

If you do keep this code here though, I think you can substantially reduce the diff, because you don't have to wrap all the code in `with override_settings(ENABLE_ENTERPRISE_INTEGRATION=enterprise_enabled):` (changing the indentation) - just the one line that actually calls the django view.